File: spring-board/service/article/src/main/java/kopher/board/article/controller/ArticleController.java
package kopher.board.article.controller;

import kopher.board.article.service.ArticleService;
import kopher.board.article.service.request.ArticleCreateRequest;
import kopher.board.article.service.request.ArticleUpdateRequest;
import kopher.board.article.service.response.ArticlePageResponse;
import kopher.board.article.service.response.ArticleResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
public class ArticleController {
    private final ArticleService articleService;

    @GetMapping("/v1/articles/{articleId}")
    public ArticleResponse read(@PathVariable Long articleId) {
        return articleService.read(articleId);
    }

    @GetMapping("/v1/articles")
    public ArticlePageResponse readAll(@RequestParam Long boardId, @RequestParam Long page, @RequestParam Long pageSize) {
        return articleService.readAll(boardId, page, pageSize);
    }

    @PostMapping("/v1/articles")
    public ArticleResponse create(@RequestBody ArticleCreateRequest request) {
        return articleService.create(request);
    }

    @PutMapping("/v1/articles/{articleId}")
    public ArticleResponse update(@PathVariable Long articleId, @RequestBody ArticleUpdateRequest request) {
        return articleService.update(articleId, request);
    }

    @DeleteMapping("/v1/articles/{articleId}")
    public void delete(@PathVariable Long articleId) {
        articleService.delete(articleId);
    }
}

File: board/service/article/src/main/kotlin/jp/co/kopher/board/article/controller/ArticleController.kt
package jp.co.kopher.board.article.controller

import jp.co.kopher.board.article.service.ArticleService
import jp.co.kopher.board.article.service.request.ArticleCreateRequest
import jp.co.kopher.board.article.service.request.ArticleUpdateRequest
import jp.co.kopher.board.article.service.response.ArticleResponse
import org.springframework.web.bind.annotation.*

@RestController
class ArticleController(
    private val articleService: ArticleService,
) {

    @GetMapping("/v1/articles/{articleId}")
    fun read(@PathVariable articleId: Long): ArticleResponse {
        return articleService.read(articleId)
    }

    @PostMapping("/v1/articles")
    fun create(@RequestBody request: ArticleCreateRequest): ArticleResponse {
        return articleService.create(request)
    }

    @PutMapping("/v1/articles/{articleId}")
    fun update(@PathVariable articleId: Long, @RequestBody request: ArticleUpdateRequest): ArticleResponse {
        return articleService.update(articleId, request)
    }

    @DeleteMapping("/v1/articles/{articleId}")
    fun delete(@PathVariable articleId: Long) {
        articleService.delete(articleId)
    }
}